common/persistence/serialization/serializer.go

```diff
@@ -527,6 +530,15 @@ func (t *serializerImpl) ReplicationTaskFromBlob(data *commonpb.DataBlob) (*repl
 	return result, ProtoDecodeBlob(data, result)
 }
 
+func (t *serializerImpl) NexusIncomingServiceToBlob(service *persistencespb.NexusIncomingService, encodingType enumspb.EncodingType) (*commonpb.DataBlob, error) {
+	return ProtoEncodeBlob(service, encodingType)
+}
+
+func (t *serializerImpl) NexusIncomingServiceFromBlob(data *commonpb.DataBlob) (*persistencespb.NexusIncomingService, error) {
+	result := &persistencespb.NexusIncomingService{}
+	return result, ProtoDecodeBlob(data, result)
+}
```

Comment:
I totally understand you wanting to keep current patterns but I'd totally support making a generic data blob to and from T conversion method.

Reply:
Yeah, I just didn't have the time and the fact that `Serializer` uses an interface which makes the implementation trickier than other generic utilities I've added. If I'll have time I'll circle back to it.